src/main/java/com/team4/testingsystem/repositories/ContentFilesRepository.java

```diff
@@ -19,4 +22,12 @@
     @Modifying
     @Query(value = "DELETE from ContentFile cf WHERE cf.id = ?1")
     int removeById(Long id);
+
+    @Query("SELECT cf from ContentFile cf " +
+           "JOIN cf.questions q " +
+           "JOIN q.level l " +
+           "WHERE l.name = ?1 " +
+           "order by function('random') ")
+    List<ContentFile> getRandomFiles(String level, Pageable pageable);
```

Comment:
why do we need a list of random files?

Reply:
BTW, does the pagination works on CrudRepository extention with queries without its direct usage